sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -1259,57 +1259,37 @@ private[lf] object SBuiltinFun {
-                          coid,
-                          dstTplId,
-                          dstArg,
-                          allowCatchingContractInfoErrors = false,
-                        ) { dstContract =>
-                          // If the destination and src templates are the same, we skip the computation
-                          // of the destination template's view and the validation of the contract info.
-                          if (dstTplId == srcTplId)
-                            k(SAny(Ast.TTyCon(dstTplId), dstArg))
-                          else
-                            checkContractUpgradable(coid, srcContract, dstContract) { () =>
-                              executeExpression(machine, SEPreventCatch(dstView)) { dstViewValue =>
-                                if (srcViewValue != dstViewValue) {
-                                  Control.Error(
-                                    IE.Upgrade(
-                                      IE.Upgrade.ViewMismatch(
-                                        coid,
-                                        interfaceId,
-                                        srcTplId,
-                                        dstTplId,
-                                        srcView = srcViewValue.toUnnormalizedValue,
-                                        dstView = dstViewValue.toUnnormalizedValue,
-                                      )
-                                    )
-                                  )
-                                } else
-                                  k(SAny(Ast.TTyCon(dstTplId), dstArg))
-                              }
-                            }
+      resolvePackageName(machine, pkgName) { pkgId =>
+        val dstTplId = srcTplId.copy(packageId = pkgId)
+        machine.ensurePackageIsLoaded(
+          dstTplId.packageId,
+          language.Reference.Template(dstTplId),
+        ) { () =>
+          ensureTemplateImplementsInterface(machine, interfaceId, coid, dstTplId) {
+            fromInterface(machine, srcTplId, srcArg, dstTplId) {
+              case None =>
+                Control.Error(IE.WronglyTypedContract(coid, dstTplId, srcTplId))
+              case Some(dstArg) =>
+                viewInterface(machine, interfaceId, dstTplId, dstArg) { dstView =>
+                  getContractInfo(
+                    machine,
+                    coid,
+                    dstTplId,
+                    dstArg,
+                    allowCatchingContractInfoErrors = false,
+                  ) { dstContract =>
+                    // If the destination and src templates are the same, we skip the computation
+                    // of the destination template's view and the validation of the contract info.
+                    if (dstTplId == srcTplId)
+                      k(SAny(Ast.TTyCon(dstTplId), dstArg))
+                    else
+                      checkContractUpgradable(coid, srcContract, dstContract) { () =>
```

Comment:
@dylant-da : thx for the updates that remove the view check.

Question on the order of checks: I'd expect that the upgradeability is checked before computing the view, for two reasons:
1. the errors are likely better
2. the ugprading check might later be extended with checks that establish invariants exploited by the interface view computation. 

Any particular reason to not do the upgrading check before the view computation?

Reply:
Sorry, in my last comment above I misspoke - the call to `viewInterface` above only retrieves the interface expression - it doesn't actually get executed. It shouldn't ever fail because we run `ensureTemplateImplementsInterface` just before it.

We are doing the upgradeability check before *executing* the view:

```scala
checkContractUpgradable(coid, srcContract, dstContract) { () => // check upgradeability
  executeExpression(machine, SEPreventCatch(dstView)) { _ => // actually calculate view
    k(SAny(Ast.TTyCon(dstTplId), dstArg))
  }
}
```

That said, given your comment below this point may be moot - if we don't calculate the view, its ordering relative to upgradeability checks is moot.